server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/RemoveRegexMappingRoutesTest.java

```diff
@@ -0,0 +1,131 @@
+
+    private WebAdminServer webAdminServer;
+    private MemoryRecipientRewriteTable memoryRecipientRewriteTable;
+
+    @BeforeEach
+    void beforeEach() throws Exception {
+        DNSService dnsService = mock(DNSService.class);
+        DomainList domainList = new MemoryDomainList(dnsService);
+        memoryRecipientRewriteTable = new MemoryRecipientRewriteTable();
+        memoryRecipientRewriteTable.setDomainList(domainList);
+        domainList.addDomain(Domain.of("domain.tld"));
+
+        webAdminServer = WebAdminUtils
+            .createWebAdminServer(new RemoveRegexMappingRoutes(memoryRecipientRewriteTable))
+            .start();
+
+        RestAssured.requestSpecification = WebAdminUtils.buildRequestSpecification(webAdminServer)
+            .setBasePath(RemoveRegexMappingRoutes.BASE_PATH)
+            .log(LogDetail.METHOD)
+            .build();
+    }
+
+    @AfterEach
+    void stop() {
+        webAdminServer.destroy();
+    }
+
+    @Test
+    void removeRegexMappingShouldReturnNoContentWhenSuccess() throws Exception {
+        MappingSource mappingSource = MappingSource.fromUser(User.fromUsername("abc@domain.tld"));
+        memoryRecipientRewriteTable
+            .addRegexMapping(mappingSource, "^[aeiou]");
+
+        with()
+            .delete("abc@domain.tld/targets/^[aeiou]")
+        .then()
+            .statusCode(HttpStatus.NO_CONTENT_204)
+            .contentType(ContentType.JSON);
+
+        assertThat(memoryRecipientRewriteTable.getStoredMappings(mappingSource))
+            .doesNotContain(Mapping.regex("^[aeiou]"));
+    }
+
+    @Test
+    void removeRegexMappingShouldAllowUserWithoutDomain() throws Exception {
+        MappingSource mappingSource = MappingSource.fromUser(User.fromUsername("abcdomaintld"));
+        memoryRecipientRewriteTable
+            .addRegexMapping(mappingSource, "^[aeiou]");
+
+        with()
+            .delete("abcdomaintld/targets/^[aeiou]")
+        .then()
+            .statusCode(HttpStatus.NO_CONTENT_204)
+            .contentType(ContentType.JSON);
+
+        assertThat(memoryRecipientRewriteTable.getStoredMappings(mappingSource))
+            .doesNotContain(Mapping.regex("^[aeiou]"));
+    }
+
+    @Test
+    void removeRegexMappingShouldReturnNotFoundWhenSourceIsEmpty() {
+        with()
+            .delete("/targets/^[aeiou]")
+        .then()
+            .statusCode(HttpStatus.NOT_FOUND_404)
+            .contentType(ContentType.JSON);
+    }
+
+    @Test
+    void removeRegexMappingShouldReturnNotFoundWhenRegexIsEmpty() {
+         with()
+            .delete("abc@domain.tld/targets/")
+        .then()
+            .statusCode(HttpStatus.NOT_FOUND_404)
+            .contentType(ContentType.JSON);
+    }
+
+    @Test
+    void removeRegexMappingShouldReturnNotFoundWhenSourceAndRegexEmpty() {
+        with()
+            .delete("/targets/")
+            .then()
+            .statusCode(HttpStatus.NOT_FOUND_404)
+            .contentType(ContentType.JSON);
+    }
```

Comment:
I need a test of  `removeRegexMappingShouldReturnBadRequestWhenInvalidRegex`

Reply:
The method will be implemented in the next commit.